core/controller/src/main/scala/org/apache/openwhisk/core/loadBalancer/ShardingContainerPoolBalancer.scala

```diff
@@ -298,7 +337,12 @@ class ShardingContainerPoolBalancer(
                               instance: InvokerInstanceId): Future[Either[ActivationId, WhiskActivation]] = {
 
     totalActivations.increment()
-    totalActivationMemory.add(action.limits.memory.megabytes)
+    if (action.exec.pull) {
```

Comment:
Can we pull this into a `isBlackbox` value and use below again? Makes it a bit more readable.